### PR TITLE
P2P peer broadcast and discovery

### PR DESCRIPTION
**Pull request by mrengstrom**
adding mDNS broadcaster to the http proxy extension

### CODE REVIEW

File: gocat-extensions/proxy/proxy_receiver_http.go [hunk 1]
```diff
@@ -292,4 +296,16 @@ func (h *HttpReceiver) initializeReceiverPort() error {
...
 	rand.Seed(time.Now().UnixNano())
 	return rand.Intn(maxPort - minPort) + minPort
-}
\ No newline at end of file
+}
+
+func (h *HttpReceiver) broadcastReceiverChannel(port int) {
+    for h.agentPaw == "" {time.Sleep(1)}
+    info := []string{"HTTP"}
+//     info = h.urlList
```
Comment (unkempthenry) [inline gocat-extensions/proxy/proxy_receiver_http.go:304]: extraneous comment?

**Review by unkempthenry — CHANGES_REQUESTED** [review]
What are your thoughts on adding?: 

- a variation(s) to the sandcat deploy ability that has the headers and options to get this running
- supporting something like a `-server null` argument to force the agent to use p2p

File: gocat-extensions/proxy/proxy_receiver_http.go (hunk 1 above)
Comment (mrengstrom) [reply gocat-extensions/proxy/proxy_receiver_http.go:304]: yup, looks like I missed that.  Will remove.

**Review by unkempthenry — APPROVED** [review]

**Comment by unkempthenry** [comment]
depends on https://github.com/mitre/gocat/pull/33

**Pull request merged**